src/pybind/mgr/dashboard/controllers/nvmeof.py

```diff
@@ -392,23 +395,148 @@ def list(self, nqn: str):
+                    http_status_code=NVMeoFClient().NVMeoFError2HTTP.get(response.status, 400),
+                    component="nvmeof",
+                )
+            # Create a new response with initiators information appended
+            for subs in subsystems.subsystems:
+                res = {
+                    'nqn': subs.nqn,
+                    'enable_ha': subs.enable_ha,
+                    'serial_number': subs.serial_number,
+                    'model_number': subs.model_number,
+                    'min_cntlid': subs.min_cntlid,
+                    'max_cntlid': subs.max_cntlid,
+                    'namespace_count': subs.namespace_count,
+                    'max_namespaces': subs.max_namespaces,
+                    'subtype': subs.subtype,
+                    'initiators': [],
+                }
+                # Get list of initiators present on the subsystem
+                hosts = NVMeoFHost().list(subs.nqn)
+                # Decode the bytes object to json
+                json_str = hosts.decode('utf-8')
+                json_obj = json.loads(json_str)
+                # Appending initiators information
+                res['initiators'] = json_obj
+                response.append(res)
+            return response
+
+        @Endpoint('POST', '/subsystem')
+        @empty_response
+        @handle_nvmeof_error
+        @CreatePermission
+        def create(self, nqn: str, enable_ha: bool,
+                   max_namespaces: int = 256, initiators: str = None):
+            subsystem_response = NVMeoFClient().stub.create_subsystem(
+                NVMeoFClient.pb2.create_subsystem_req(
+                    subsystem_nqn=nqn, max_namespaces=max_namespaces, enable_ha=enable_ha
+                )
+            )
+
+            if subsystem_response.status != 0:
+                return subsystem_response
+
+            if initiators is not None:
+                all_host_nqns = initiators.split(',')
+                for host_nqn in all_host_nqns:
+                    add_host_response = NVMeoFClient().stub.add_host(
+                        NVMeoFClient.pb2.add_host_req(subsystem_nqn=nqn, host_nqn=host_nqn)
+                    )
+                    if add_host_response.status != 0:
+                        return add_host_response
+
+            return subsystem_response
+
+        @staticmethod
+        def determine_nqns(res, new_host_nqns):
+            to_delete_nqns = []
+            to_add_nqns = []
+
+            current_host_nqns = [host.nqn for host in res.hosts]
+
+            # All hosts are allowed but request is to add new hosts
+            if res.allow_any_host and new_host_nqns != ["*"]:
+                to_delete_nqns = ["*"]
+                to_add_nqns = new_host_nqns
+
+            # All hosts are not allowed but request is to allow all
+            elif new_host_nqns == ["*"]:
+                to_delete_nqns = current_host_nqns
+                to_add_nqns.append("*")
```

Comment:
Why? Even if it doesn't (apparently) make much sense, the CLI supports adding both `*` and specific hosts NQNs. It's common with access rules that you can add contradictory rules like the following:
```
allow all
allow user X do that
```
The main reason for that is that in some situations you just want to temporarily grant extra permissions (troubleshooting, etc.) without deleting the existing rulesets.

Reply:
If both are allowed then lets do the same from UI too, makes sense 👍🏿